bokehjs/src/lib/client/session.ts

```diff
@@ -44,10 +44,14 @@ export class ClientSession {
     }
   }
 
+  // TODO: refactor to connection_lost_permanently
   notify_connection_lost(): void {
     this.document.event_manager.send_event(new ConnectionLost())
   }
 
+  // TODO: notify_connection_retry ?
+  // data: time (ms) to next retry, connection attempt number
+
```

Comment:
I will implement appropriate mechanisms separately from this PR (issues https://github.com/bokeh/bokeh/issues/14252 and https://github.com/bokeh/bokeh/issues/6884 for tracking).

Reply:
I think a new `DocumentEvent` is a perfectly fine start here and then people can set up their own notifications and we can hook up a built-in notification system later.